pkg/disruption/backend/logger/logger.go

```diff
@@ -0,0 +1,63 @@
+package logger
+
+import (
+	"github.com/openshift/origin/pkg/disruption/backend"
+	backendsampler "github.com/openshift/origin/pkg/disruption/backend/sampler"
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+
+	"github.com/sirupsen/logrus"
+)
+
+// NewLogger returns a new instance of SampleCollector that logs samples
+func NewLogger(delegate backendsampler.SampleCollector, name string, connType monitorapi.BackendConnectionType) *logger {
+	return &logger{
+		delegate: delegate,
+		name:     name,
+		connType: connType,
+		logAll:   false,
+	}
+}
+
+type logger struct {
+	delegate backendsampler.SampleCollector
+	name     string
+	connType monitorapi.BackendConnectionType
+	logAll   bool
+}
+
+func (l logger) Collect(s backend.SampleResult) {
+	// we receive sample in ordered sequence, 1, 2, ... n
+	if l.delegate != nil {
+		l.delegate.Collect(s)
+	}
+	l.log(s)
+}
+
+func (l logger) log(s backend.SampleResult) {
+	if s.Sample == nil {
+		return
+	}
+
+	err := s.AggregateErr()
+	rr := s.RequestResponse
+	_, retry := rr.IsRetryAfter()
+	// we will log it only when it has interesting data,
+	// an error, a retry after, or shutdown window in progress
+	if !(l.logAll || (err != nil || rr.ShutdownInProgress() || retry)) {
+		return
+	}
+
+	entry := logrus.WithFields(logrus.Fields{
+		"backend": l.name,
+		"type":    l.connType,
+		"sample":  s.Sample.ID,
+	})
+	entry = entry.WithFields(s.Fields())
+
+	switch {
+	case err != nil:
+		entry.Errorf("failure: %v", err)
+	default:
+		entry.Infof("received a sample")
```

Comment:
Is this temporary or long lived? Looks like about 1900 lines in the job I was looking at, not outlandish, but curious what the plan is.

Reply:
Just throwing around an idea here, but a separate logfile could be interesting here. We could get it pushed to loki, giving us the ability to look for failed disruption samples globally across CI during windows of time. Not anything needed at this point in time but something we can think about for the future.

Reply:
this logs ineresting events only, which are any error, or a response with shutdown in progress or a retry-after, it can come in handy if we were to debug the disruption, i like the idea of a separate log file